django/http/request.py

```diff
@@ -89,14 +90,48 @@ def headers(self):
 
     @cached_property
     def accepted_types(self):
-        """Return a list of MediaType instances."""
-        return parse_accept_header(self.headers.get("Accept", "*/*"))
+        """Return a list of MediaType instances, in order of preference."""
+        header_value = self.headers.get("Accept", "*/*")
+        return sorted(
+            (MediaType(token) for token in header_value.split(",") if token.strip()),
+            key=operator.attrgetter("quality", "specificity"),
+            reverse=True,
+        )
 
-    def accepts(self, media_type):
-        return any(
-            accepted_type.match(media_type) for accepted_type in self.accepted_types
+    def accepted_type(self, media_type):
+        """
+        Return the most preferred MediaType instance which matches the given media type
```

Comment:
```suggestion
        Return the preferred MediaType instance which matches the given media
        type.
```

Throughout, `preferred` without the `most`.